### PR TITLE
split the provider interface

### PR DESCRIPTION
**Pull request by Liujingfang1**
This is the change we need to do to work with the cli-utils change in https://github.com/kubernetes-sigs/cli-utils/pull/270.

### CODE REVIEW

File: pkg/live/dual-delegating-provider.go [hunk 1]
```diff
@@ -50,98 +39,29 @@ func (cp *DualDelegatingProvider) Factory() util.Factory {
...
-// CachedManifestReader implements ManifestReader, storing objects to return.
-type CachedManifestReader struct {
-	objs []*unstructured.Unstructured
-}
-
-// Read simply returns the stored objects.
-func (r *CachedManifestReader) Read() ([]*unstructured.Unstructured, error) {
-	return r.objs, nil
+	return inventory.NewInventoryClient(cp.Factory(),
+		inventroyWrapperFunc,
```
Comment (seans3) [inline pkg/live/dual-delegating-provider.go:43]: inventroy -> inventory ?

File: pkg/live/dual-delegating-provider.go [hunk 2]
```diff
@@ -50,98 +39,29 @@ func (cp *DualDelegatingProvider) Factory() util.Factory {
...
-// findResourceGroupInv returns the pointer to the ResourceGroup inventory object,
-// or nil if it does not exist.
-func findResourceGroupInv(objs []*unstructured.Unstructured) *unstructured.Unstructured {
-	for _, obj := range objs {
-		if inventory.IsInventoryObject(obj) {
-			if obj.GetKind() == "ResourceGroup" {
-				return obj
-			}
-		}
+func inventroyWrapperFunc(obj *unstructured.Unstructured) inventory.Inventory {
```
Comment (seans3) [inline pkg/live/dual-delegating-provider.go:47]: inventroyWrapperFunc -> inventory ?

File: go.mod [hunk 3]
```diff
@@ -26,3 +26,5 @@ require (
 	sigs.k8s.io/kustomize/cmd/config v0.8.5
 	sigs.k8s.io/kustomize/kyaml v0.9.4
 )
+
+replace sigs.k8s.io/cli-utils => ../../sigs.k8s.io/cli-utils
```
Comment (seans3) [inline go.mod:30]: After cli-utils is submitted, this will have to be updated and removed.

File: pkg/live/dual-delegating-provider.go (hunk 1 above)
Comment (Liujingfang1) [reply pkg/live/dual-delegating-provider.go:43]: Oops, will fix it.

File: go.mod (hunk 3 above)
Comment (Liujingfang1) [reply go.mod:30]: Yep, I'll remove this after the PR kubernetes-sigs/cli-utils#270 is merged.

File: pkg/live/dual-delegating-loader.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+	}
+	if rgInv != nil && cmInv != nil {
+		return nil, objs, MultipleInventoryObjError{
+			InvObjs: []*unstructured.Unstructured{rgInv, cmInv},
+		}
+	}
+	if cmInv != nil {
+		inv = inventory.WrapInventoryInfoObj(cmInv)
+	}
+	//cp.calcInventory = true
```
Comment (seans3) [inline pkg/live/dual-delegating-loader.go:115]: We should probably remove this.

**Review by seans3 — COMMENTED** [review]
some small nits

File: pkg/live/dual-delegating-provider.go (hunk 1 above)
Comment (Liujingfang1) [reply pkg/live/dual-delegating-provider.go]: fixed

File: pkg/live/dual-delegating-provider.go (hunk 2 above)
Comment (Liujingfang1) [reply pkg/live/dual-delegating-provider.go]: fixed

File: go.mod (hunk 3 above)
Comment (Liujingfang1) [reply go.mod]: kubernetes-sigs/cli-utils#270 is merged. I published a new version for cli-utils: v0.22.0 and used it in `go.mod`.

File: pkg/live/dual-delegating-loader.go (hunk 4 above)
Comment (Liujingfang1) [reply pkg/live/dual-delegating-loader.go]: Done.

**Review by seans3 — APPROVED** [review]
/lgtm

**Pull request merged**